src/bentoml/_internal/models/model.py

```diff
@@ -606,8 +606,9 @@ def with_options(self, **kwargs: t.Any) -> ModelInfo:
     def imported_module(self) -> ModuleType:
         if self._cached_module is None:
             try:
+                module_to_import = self.module or "bentoml.dummy_module"
                 object.__setattr__(
-                    self, "_cached_module", importlib.import_module(self.module)
+                    self, "_cached_module", importlib.import_module(module_to_import)
```

Comment:
I don't think we should do this.

Reply:
@aarnphm  there's some codes in other places relies on this `self.imported_module` has certain functions. We have a `dummy_module` to implement a dummy version of these functions so that we are not distinguishing our framework codes and users' codes.

Reply:
Then I think we should fix this. I think `_cached_module` is a hack.

Reply:
I'm confused by what you mean by 'fix this'. Do we not want `_cached_module` or `imported_module`? I think those are needed for other functionalities.

Reply:
Another way to avoid this empty module error (right now, `self.module` is "" due to the nature of the `modules.save` method) is to have `self.imported_module` return None if `self_module == ""`. Then in all other methods that use `self.imported_module`, we add a check for the value of `self.imported_module`.

The current approach we are taking is to add a dummy_module that imitates all the functionalities that an actual module will have. This way, other methods that use the `imported_module` only need to worry about how they interact with modules and what the modules are.

Reply:
I think we can just raise an error here if this method is called at all for a model saved without a module.

Reply:
@sauyon This `import_module` is somehow called when bentoml push a model to BentoCloud so we need to avoid raising the error here.

Reply:
We should probably just remove the use in push?

Reply:
Following the error traceback of calling `bentoml.models.push`, it appears that the error is triggered when we call the `options` property of the `ModelInfo` class. This line: https://github.com/bentoml/BentoML/blob/main/src/bentoml/_internal/cloud/bentocloud.py#L639C44-L639C44

Accessing the `options` will call `imported_module` under the hood. I don't think we can easily avoid this error in push.

Reply:
I think it just seems like more effort than necessary when we can just patch `options`?

Reply:
I think we should error for both of those. Honestly I would want to error for `options` as well, but we'd have to figure something else out for push and that seems more annoying.